src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -220,6 +222,30 @@ public static boolean isCorrectToolForDrops(@Nonnull BlockState state, @Nonnull
         return player.hasCorrectToolForDrops(state);
     }
 
+    public static Pair<Predicate<UseOnContext>, Consumer<UseOnContext>> getHoeTillable(UseOnContext context, Pair<Predicate<UseOnContext>, Consumer<UseOnContext>> pair)
```

Comment:
I really don't like this.
In my opinion there are two cleaner solutions to this:

1. Add more parameters to `getToolModifiedState` so that it can construct the `UseOnContext` and use the native methods in `HoeItem` like it does for all other actions.
2. Change `getToolModifiedState` to take a `UseOnContext` instead of separate parameters (might be limiting, but a `UseOnContext` is just a POJO...) and then do the same as in 1.
3. Change the event from being cancellable to having a result so we can distinguish "cancel action" and "use vanilla". `getToolModifiedState` would need to have its return type altered as well then.

In my opinion 1 or 2 are the best options here.